src/Type/ErrorType.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+class ErrorType implements Type
```

Comment:
The rules can check for ErrorType explicitly, but it usually should be enough to rely on the yes/maybe/no system in union/intersection types (not really usable at this point, but should work well when more stuff from #192 is implemented.

For example type of `$foo` at the end of the following code should be `DateTime | Error(undefined variable)`.

~~~php
if (randomBool()) {
     $foo = new DateTime();
}
~~~

If you then call `$foo->format()` then rule on low level can emit error if `$fooType->isMethodCallable('format') === NO` which the code would pass (as the method may be called on the `$foo` variable).

However the same rule on higher level can emit error if  `$fooType->isMethodCallable('format') !== YES` which the code would not pass as error „Unable to call format() on type of DateTime | Error(undefined variable)“ would be emitted.


---

Currently the ErrorType has very limited use as it's only used when using foreach with value that is not iterable. For example the following code results in error „Cannot call method format() on *ERROR*“.

~~~php
foreach (123 as $value) {
	$value->format('...');
}
~~~